tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/record/ClosedRecordTest.java

```diff
@@ -323,4 +323,28 @@ public void testInvalidExprsAsRecordLiteralKeys() {
         BAssertUtil.validateError(result, index++, "undefined field 'z' in record 'Foo'", 46, 48);
         Assert.assertEquals(result.getErrorCount(), index);
     }
+
+    @Test
+    public void testOptionalRecordRemove() {
+        BValue[] returns = BRunUtil.invoke(compileResult, "removeOptional");
+
+        Assert.assertTrue(((BBoolean) returns[0]).booleanValue());
```

Comment:
Shall we do these assertions in Ballerina itself?

Reply:
Will do